internal/uvm/pipes.go

```diff
@@ -54,19 +56,62 @@ func (uvm *UtilityVM) RemovePipe(ctx context.Context, hostPath string) error {
 	return nil
 }
 
-// IsPipe returns true if the given path references a named pipe.
+type NamedPipe struct {
+	HostPath      string
+	ContainerPath string
+	UVMPipe       bool
+}
+
+// IsPipe returns true if the given path references a named pipe. The pipe can be:
+// - host named pipe shared via VSMB
+// - UVM named pipe
 func IsPipe(hostPath string) bool {
+	if pipePath, ok := strings.CutPrefix(hostPath, guestpath.UVMMountPrefix); ok {
+		return strings.HasPrefix(pipePath, pipePrefix)
+	}
 	return strings.HasPrefix(hostPath, pipePrefix)
 }
 
-// GetContainerPipeMapping returns the source and destination to use for a given
-// pipe mount in a container.
-func GetContainerPipeMapping(uvm *UtilityVM, mount specs.Mount) (src string, dst string) {
-	if uvm == nil {
-		src = mount.Source
-	} else {
-		src = vsmbSharePrefix + `IPC$\` + strings.TrimPrefix(mount.Source, pipePrefix)
+// ParseNamedPipe tries parses the mount as a named pipe and returns `NamedPipe` mapping
+// for a container.
+// The pipe mount can be either a host pipe shared via VSMB or a UVM pipe.
+func ParseNamedPipe(uvm *UtilityVM, mount specs.Mount) (NamedPipe, bool) {
+	// destination is not a named pipe, return false
+	if !strings.HasPrefix(mount.Destination, pipePrefix) {
+		return NamedPipe{}, false
+	}
+
+	// host pipe mapped into container either for process isolated containers or
+	// for Hyper-V isolated WCOW over VSMB
+	isHostPipe := false
+	// UVM pipe mapped into container
+	isUVMPipe := false
+	sourcePath := mount.Source
+	containerPath := strings.TrimPrefix(mount.Destination, pipePrefix)
+	if strings.HasPrefix(mount.Source, pipePrefix) {
+		isHostPipe = true
+		if uvm != nil {
+			sourcePath = vsmbSharePrefix + `IPC$\` + strings.TrimPrefix(mount.Source, pipePrefix)
+		}
+	} else if strings.HasPrefix(mount.Source, guestpath.UVMMountPrefix) {
+		sourcePath = strings.TrimPrefix(mount.Source, guestpath.UVMMountPrefix)
+		// check if it's a UVM pipe
+		if strings.HasPrefix(sourcePath, pipePrefix) {
+			isUVMPipe = true
+			podID := strings.TrimSuffix(uvm.id, "@vm")
+			sourcePath = strings.TrimPrefix(sourcePath, pipePrefix)
+			// TODO (anmaxvl): LCOW doesn't support UVM named pipes at the moment
+			if uvm.operatingSystem == "windows" {
```

Comment:
since Linux named pipes are just files, without the `\\.\pipe\` path prefix, and L80-83 check if the destination has that prefix, can we assume this code path is for Windows only, and move this to the start of this function?